server/sonar-server/src/main/java/org/sonar/server/measure/index/ProjectMeasuresIndex.java

```diff
@@ -163,6 +175,56 @@ public ProjectMeasuresIndex(EsClient client, AuthorizationTypeSupport authorizat
     return new SearchIdResult<>(requestBuilder.get(), id -> id);
   }
 
+  public ProjectMeasuresStatistics searchTelemetryStatistics() {
+    SearchRequestBuilder request = client
+      .prepareSearch(INDEX_TYPE_PROJECT_MEASURES)
+      .setFetchSource(false)
+      .setSize(0);
+
+    BoolQueryBuilder esFilter = boolQuery();
+    request.setQuery(esFilter);
+    request.addAggregation(AggregationBuilders.terms(FIELD_LANGUAGES)
+      .field(FIELD_LANGUAGES)
+      .size(MAX_PAGE_SIZE)
+      .minDocCount(1)
+      .order(Terms.Order.count(false)));
+    request.addAggregation(AggregationBuilders.nested(FIELD_NCLOC_LANGUAGE_DISTRIBUTION, FIELD_NCLOC_LANGUAGE_DISTRIBUTION)
+      .subAggregation(AggregationBuilders.terms(FIELD_NCLOC_LANGUAGE_DISTRIBUTION + "_terms")
+        .field(FIELD_DISTRIB_LANGUAGE)
+        .size(MAX_PAGE_SIZE)
+        .minDocCount(1)
+        .order(Terms.Order.count(false))
+        .subAggregation(sum(FIELD_DISTRIB_NCLOC).field(FIELD_DISTRIB_NCLOC))));
+
+    Stream.of(LINES_KEY, NCLOC_KEY)
+      .forEach(metric -> request.addAggregation(AggregationBuilders.nested(metric, FIELD_MEASURES)
+        .subAggregation(AggregationBuilders.filter(metric + "_filter", termQuery(FIELD_MEASURES_KEY, metric))
+          .subAggregation(sum(metric + "_filter_sum").field(FIELD_MEASURES_VALUE)))));
+
+    ProjectMeasuresStatistics.Builder statistics = ProjectMeasuresStatistics.builder();
+
+    SearchResponse response = request.get();
+    statistics.setProjectCount(response.getHits().getTotalHits());
+    Stream.of(LINES_KEY, NCLOC_KEY)
+      .map(metric -> (Nested) response.getAggregations().get(metric))
+      .map(nested -> (Filter) nested.getAggregations().get(nested.getName() + "_filter"))
+      .map(filter -> (Sum) filter.getAggregations().get(filter.getName() + "_sum"))
+      .forEach(sum -> {
+        String metric = sum.getName().replace("_filter_sum", "");
+        long value = Math.round(sum.getValue());
+        statistics.setSum(metric, value);
+      });
+    statistics.setProjectCountByLanguage(termsToMap(response.getAggregations().get(FIELD_LANGUAGES)));
+    Function<Terms.Bucket, Long> bucketToNcloc = bucket -> Math.round(((Sum) bucket.getAggregations().get(FIELD_DISTRIB_NCLOC)).getValue());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "Sum". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1905)